juju_verify/verifiers/ovn_central.py

```diff
@@ -0,0 +1,397 @@
+            "`cluster-status` action manually."
+        )
+        if not self._complete_cluster_status:
+            action_results = run_action_on_units(
+                self.all_application_units, "cluster-status", use_cache=False
+            )
+            for unit_name, result in action_results.items():
+                sb_data = data_from_action(result, "southbound-cluster")
+                nb_data = data_from_action(result, "northbound-cluster")
+                if not sb_data:
+                    raise JujuVerifyError(report_err.format(unit_name, "Southbound"))
+                if not nb_data:
+                    raise JujuVerifyError(report_err.format(unit_name, "Northbound"))
+
+                self._complete_cluster_status[unit_name] = UnitClusterStatus(
+                    southbound=ClusterStatus(sb_data), northbound=ClusterStatus(nb_data)
+                )
+
+        return self._complete_cluster_status
+
+    @property
+    def all_application_units(self) -> List[Unit]:
+        """Return list of all units from an application of currently verified unit(s)."""
+        if not self._all_application_units:
+            app = self.units[0].application
+            self._all_application_units = [
+                unit for unit in self.model.units.values() if unit.application == app
+            ]
+        return self._all_application_units
+
+    @staticmethod
+    def cluster_tolerance(size: int) -> int:
+        """Return number of failed nodes that the cluster can tolerate.
+
+        This calculation is based on the raft protocol's requirement to have "(N/2) + 1"
+        active nodes to maintain quorum.
+
+        :param size: Current size of a cluster.
+        :return: Number of nodes that can fail without loosing cluster quorum
+        """
+        if size < 1:
+            return 0
+
+        min_quorum_size = (size // 2) + 1
+        return size - min_quorum_size
+
+    def check_single_application(self) -> Result:
+        """Verify that all units that are being verified belong to the same app."""
+        apps = {unit.application for unit in self.units}
+        if len(apps) > 1:
+            app_list = ", ".join(apps)
+            result = Result(
+                Severity.FAIL,
+                f"Can't verify multiple ovn-central application at the same time. "
+                f"Currently selected units belong to: {app_list}",
+            )
+        else:
+            result = Result(
+                Severity.OK, "Selected units are part of only one application."
+            )
+
+        return result
+
+    def check_leader_consistency(self) -> Result:
+        """Verify consistency of ovn-central cluster."""
+        result = Result()
+        leader_mappings: Dict[str, Dict[str, List[str]]] = {
+            "Southbound": defaultdict(list),
+            "Northbound": defaultdict(list),
+        }
+        for unit, cluster_data in self.complete_cluster_status.items():
+            clusters = [
```

Comment:
It seems we could put clusters outside the for loop. We only initial it one time. what do you think?